### ISSUE TITLE
Error building/using CGAL

### ISSUE THREAD

**Issue by JackBoosY** [CGAL/cgal#4286]
Hi Developer,
Recently I tried to update the cgal version in vcpkg and fix the usage issue([#8571](https://github.com/microsoft/vcpkg/pull/8571)). But I have encountered some difficulties.

## Issue Details
1. Cannot build with Visual Studio 2019.
2. Cannot use `find_package(CGAL CONFIG)`
3. The usage generated by cmake has lost `COMPONENTS`.
4. Why use `CGAL::CGAL_Core` need to reconfigure the cmake file?

## Source Code
```
find_package(CGAL CONFIG COMPONENTS CGAL::CGAL_Core CGAL::CGAL_ImageIO REQUIRED)
target_link_libraries(TestProject PRIVATE CGAL::CGAL CGAL::CGAL_Core CGAL::CGAL_ImageIO TBB::${TARGET_NAME}))
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Windows 10 ver 1903
* Compiler: Visual Studio 2017/ Visual Studio 2019
* Release or debug mode: Both
* Specific flags used (if any): no
* CGAL version: 4.14.1
* Boost version: 1.71.0
* Other libraries versions if used (Eigen, TBB, etc.):
Eigen: 3.3.7
TBB: latest
qt: 5.12.5

Can you help me? Thanks.

**Comment by lrineau** [comment]
There seems to be a bug. I have seen several bug reports like that one.

Maybe it is related to https://github.com/microsoft/vcpkg/issues/8328 and our pull-request https://github.com/CGAL/cgal/pull/4285.

**Comment by JackBoosY** [comment]
@lrineau Thanks for reply. Do I need to wait for #4285 to merge and fix it?

**Comment by lrineau** [comment]
Things are unclear with that bug, for the moment. But right: I suggest you wait, if you can afford the delay.

**Comment by lrineau** [comment]
@maxGimeno Can you please try to reproduce the bug (with vcpkg on Windows), and then try again with the patch from https://github.com/CGAL/cgal/pull/4285?

**Comment by maxGimeno** [comment]
There is indeed a probel with Visual 2019, but 
`find_package(CGAL CONFIG COMPONENTS CGAL::CGAL_Core CGAL::CGAL_ImageIO REQUIRED)` is a wrong syntax. It should be
`find_package(CGAL CONFIG COMPONENTS Core ImageIO REQUIRED)`

**Comment by maxGimeno** [comment]
For usage with Visual 2019, see https://stackoverflow.com/questions/55204593/how-to-specify-cmake-version-used-by-vcpkg

The problem does not come from CGAL but from the version of cmake used by vcpkg at the time of the installation. VCPKG_FORCE_SYSTEM_BINARIES with a valid version of cmake (3.14+)  available in the PATH should make it work.

**Comment by lrineau** [comment]
@JackBoosY I suggest that you update CMake in vcpkg directly, instead of relying on system installations.

**Comment by JackBoosY** [comment]
> There is indeed a probel with Visual 2019, but
find_package(CGAL CONFIG COMPONENTS CGAL::CGAL_Core CGAL::CGAL_ImageIO REQUIRED) is a wrong syntax. It should be
find_package(CGAL CONFIG COMPONENTS Core ImageIO REQUIRED)

This issue exists not only on Visual Studio 2019, but also on Visual Studio 2017. Is this a cmake error?

**Comment by JackBoosY** [comment]
@lrineau Can #4285 works on non-header build?

**Comment by maxGimeno** [comment]
> @maxGimeno This issue exists not only on Visual Studio 2019, but also on Visual Studio 2017. Is this a cmake error?

I don't know, I don't reproduce this behaviour. I can build without problem with 2019

**Comment by lrineau** [comment]
> There is indeed a probel with Visual 2019, but
> `find_package(CGAL CONFIG COMPONENTS CGAL::CGAL_Core CGAL::CGAL_ImageIO REQUIRED)` is a wrong syntax.

Probably our `CGALConfig.cmake` should detect that, and emit an error.

**Comment by lrineau** [comment]
> @lrineau Can #4285 works on non-header build?

Yes, of course, but it will probably not fix any bug about non-header builds.

**Comment by JackBoosY** [comment]
@lrineau Well, we also need to fix it in non-header builds.

**Comment by lrineau** [comment]
@JackBoosY:
  - To support MSVC 2019, you need CMake 3.14 or later. That is actually already fixed in vcpkg, by https://github.com/microsoft/vcpkg/pull/5702. That is a vcpkg issue, and not related to CGAL.
  - The other error you reported is because of an error in your CMake code, as reported by Maxime in https://github.com/CGAL/cgal/issues/4286#issuecomment-541594162:
    > `find_package(CGAL CONFIG COMPONENTS CGAL::CGAL_Core CGAL::CGAL_ImageIO REQUIRED)` is a wrong syntax. It should be
    > `find_package(CGAL CONFIG COMPONENTS Core ImageIO REQUIRED)`

https://github.com/CGAL/cgal/pull/4285 fixes a bug in header-only mode. But as far I know, there is not any bug in the non-header-only mode, in CGAL-4.14.

**Comment by afabri** [comment]
Even if it is an error in vcpkg, shouldn't we document it as a problem in installation instructions?

**Comment by JackBoosY** [comment]
@lrineau But looking at the above details, the build was successful, but there was some problems with the use.

**Comment by JackBoosY** [comment]
@lrineau Okay, thanks.

**Comment by dim8** [comment]
I am encountering the same issue with CGAL 4.14-3 from vcpkg using Visual Studio 2017:
```
find_package(CGAL CONFIG REQUIRED)
# Note: 1 target(s) were omitted.
target_link_libraries(main PRIVATE ${cgal_lib} CGAL::${cgal_lib} CGAL_Qt5_moc_and_resources CGAL::Qt5_moc_and_resources)
```

**Comment by maxGimeno** [comment]
`${cgal_lib} CGAL::${cgal_lib} ` seems very suspicious to me.

**Comment by lrineau** [comment]
Indeed, the issue was in CGAL-4.14 as packaged by vcpkg. But it is fixed by the update to CGAL-5.0, in PR https://github.com/microsoft/vcpkg/pull/8659. Please update your local copy of vcpkg.
